### DISCUSSION TITLE
how to form a wyoming compatible http request?

### DISCUSSION THREAD

**Discussion by jereskuta** [rhasspy/piper#408] [Q&A]
Hi! I've got the wyoming-piper running in a docker container in a separate machine in my local network. Piper and whisper are working perfectly with home assistant.  

I would like to use the existing piper server in my other programs as well, but I can't get the http requests to work.
I've tried
`curl -X POST -H 'Content-Type:application/json' -d '{"text": "This is a test."}' -o test.wav 'server.ip.addr:10200'` 
`curl -G --data-urlencode "text=This is a test." -o test.wav server.ip.addr:10200`

The request gets sent but I don't recieve the audio file, and the request keeps waiting for the response.
I've tried curl GET&POST in linux and windows and neither seem to work.

Docker is running in an Ubuntu LXC in Proxmox if it makes a difference.

Has anyone got some insight on how to make this work?

**Comment by chennin** [comment]
It doesn't appear to be an HTTP API, but a TCP API, so curl won't work.

I guess we can [do what Home Assistant does in Python](https://github.com/home-assistant/core/blob/0278735dbfc4e64b146faed2e3ac3c997703e782/homeassistant/components/wyoming/tts.py#L109) or a [Golang HTTP wrapper](https://github.com/arunk140/serve-piper-tts/blob/main/Docs.md).
